pkg/compose/pull.go

```diff
@@ -63,11 +63,11 @@ func (s *composeService) pull(ctx context.Context, project *types.Project, opts
 		imagesBeingPulled = map[string]string{}
 	)
 
-	for i, service := range project.Services {
-		i, service := i, service
+	i := 0
```

Comment:
could remove this if we turned `pullErrors` from a `[]error` to a `map[string]error`, where the key is the service name?

Reply:
we can't, as `pullServiceImage` is ran in goroutines and then would required concurrent access to such a map. Here we use a pre-allocated slice with index-based writes.

anyway, I'd like to revisit in a future PR the pattern we use to access services, typically `InDependencyOrder` so that this one collects results from goroutines and return a `map[string]T`, so that we only have to manage go-concurrency in a single place